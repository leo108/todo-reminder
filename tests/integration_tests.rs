use std::process::Command;

#[test]
fn test_main_with_table_format() {
    let output = Command::new("cargo")
        .arg("run")
        .arg("tests/test_files/config.toml")
        .arg("--format=table")
        .arg("--no-tty")
        .arg("--exit-zero")
        .output()
        .expect("Failed to execute command");

    assert!(output.status.success());
    let stdout = String::from_utf8_lossy(&output.stdout);
    insta::assert_debug_snapshot!(stdout.lines().collect::<Vec<&str>>());
}

#[test]
fn test_main_with_json_format() {
    let output = Command::new("cargo")
        .arg("run")
        .arg("tests/test_files/config.toml")
        .arg("--format=json")
        .arg("--exit-zero")
        .output()
        .expect("Failed to execute command");

    assert!(output.status.success());
    let stdout = String::from_utf8_lossy(&output.stdout);
    insta::assert_debug_snapshot!(stdout.lines().collect::<Vec<&str>>());
}
