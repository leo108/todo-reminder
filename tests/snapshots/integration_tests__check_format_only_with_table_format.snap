---
source: tests/integration_tests.rs
expression: "stdout.lines().collect::<Vec<&str>>()"
---
[
    "╭──────┬────────┬──────────┬───────┬──────────────────────────╮",
    "│ Line ┆ Type   ┆ Due Date ┆ Owner ┆ bash/test.sh             │",
    "╞══════╪════════╪══════════╪═══════╪══════════════════════════╡",
    "│ 2    ┆ Format ┆          ┆       ┆ # TODO: incorrect format │",
    "╰──────┴────────┴──────────┴───────┴──────────────────────────╯",
    "╭──────┬────────┬──────────┬───────┬───────────────────────────╮",
    "│ Line ┆ Type   ┆ Due Date ┆ Owner ┆ c/test.c                  │",
    "╞══════╪════════╪══════════╪═══════╪═══════════════════════════╡",
    "│ 2    ┆ Format ┆          ┆       ┆ // TODO: incorrect format │",
    "├╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┤",
    "│ 8    ┆ Format ┆          ┆       ┆ /*                        │",
    "│      ┆        ┆          ┆       ┆ TODO: incorrect format    │",
    "│      ┆        ┆          ┆       ┆ */                        │",
    "╰──────┴────────┴──────────┴───────┴───────────────────────────╯",
    "╭──────┬────────┬──────────┬───────┬───────────────────────────╮",
    "│ Line ┆ Type   ┆ Due Date ┆ Owner ┆ c_sharp/test.cs           │",
    "╞══════╪════════╪══════════╪═══════╪═══════════════════════════╡",
    "│ 2    ┆ Format ┆          ┆       ┆ // TODO: incorrect format │",
    "├╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┤",
    "│ 8    ┆ Format ┆          ┆       ┆ /*                        │",
    "│      ┆        ┆          ┆       ┆ TODO: incorrect format    │",
    "│      ┆        ┆          ┆       ┆ */                        │",
    "╰──────┴────────┴──────────┴───────┴───────────────────────────╯",
    "╭──────┬────────┬──────────┬───────┬───────────────────────────╮",
    "│ Line ┆ Type   ┆ Due Date ┆ Owner ┆ cpp/test.cpp              │",
    "╞══════╪════════╪══════════╪═══════╪═══════════════════════════╡",
    "│ 2    ┆ Format ┆          ┆       ┆ // TODO: incorrect format │",
    "├╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┤",
    "│ 8    ┆ Format ┆          ┆       ┆ /*                        │",
    "│      ┆        ┆          ┆       ┆ TODO: incorrect format    │",
    "│      ┆        ┆          ┆       ┆ */                        │",
    "╰──────┴────────┴──────────┴───────┴───────────────────────────╯",
    "╭──────┬────────┬──────────┬───────┬──────────────────────────────╮",
    "│ Line ┆ Type   ┆ Due Date ┆ Owner ┆ css/test.css                 │",
    "╞══════╪════════╪══════════╪═══════╪══════════════════════════════╡",
    "│ 2    ┆ Format ┆          ┆       ┆ /* TODO: incorrect format */ │",
    "├╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┤",
    "│ 8    ┆ Format ┆          ┆       ┆ /*                           │",
    "│      ┆        ┆          ┆       ┆ TODO: incorrect format       │",
    "│      ┆        ┆          ┆       ┆ */                           │",
    "╰──────┴────────┴──────────┴───────┴──────────────────────────────╯",
    "╭──────┬────────┬──────────┬───────┬───────────────────────────╮",
    "│ Line ┆ Type   ┆ Due Date ┆ Owner ┆ go/test.go                │",
    "╞══════╪════════╪══════════╪═══════╪═══════════════════════════╡",
    "│ 2    ┆ Format ┆          ┆       ┆ // TODO: incorrect format │",
    "├╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┤",
    "│ 8    ┆ Format ┆          ┆       ┆ /*                        │",
    "│      ┆        ┆          ┆       ┆ TODO: incorrect format    │",
    "│      ┆        ┆          ┆       ┆ */                        │",
    "╰──────┴────────┴──────────┴───────┴───────────────────────────╯",
    "╭──────┬────────┬──────────┬───────┬───────────────────────────╮",
    "│ Line ┆ Type   ┆ Due Date ┆ Owner ┆ java/test.java            │",
    "╞══════╪════════╪══════════╪═══════╪═══════════════════════════╡",
    "│ 2    ┆ Format ┆          ┆       ┆ // TODO: incorrect format │",
    "├╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┤",
    "│ 8    ┆ Format ┆          ┆       ┆ /*                        │",
    "│      ┆        ┆          ┆       ┆ TODO: incorrect format    │",
    "│      ┆        ┆          ┆       ┆ */                        │",
    "╰──────┴────────┴──────────┴───────┴───────────────────────────╯",
    "╭──────┬────────┬──────────┬───────┬───────────────────────────╮",
    "│ Line ┆ Type   ┆ Due Date ┆ Owner ┆ javascript/test.js        │",
    "╞══════╪════════╪══════════╪═══════╪═══════════════════════════╡",
    "│ 2    ┆ Format ┆          ┆       ┆ // TODO: incorrect format │",
    "├╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┤",
    "│ 8    ┆ Format ┆          ┆       ┆ /*                        │",
    "│      ┆        ┆          ┆       ┆ TODO: incorrect format    │",
    "│      ┆        ┆          ┆       ┆ */                        │",
    "╰──────┴────────┴──────────┴───────┴───────────────────────────╯",
    "╭──────┬────────┬──────────┬───────┬───────────────────────────╮",
    "│ Line ┆ Type   ┆ Due Date ┆ Owner ┆ php/test.php              │",
    "╞══════╪════════╪══════════╪═══════╪═══════════════════════════╡",
    "│ 4    ┆ Format ┆          ┆       ┆ // TODO: incorrect format │",
    "├╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┤",
    "│ 10   ┆ Format ┆          ┆       ┆ /*                        │",
    "│      ┆        ┆          ┆       ┆ TODO: incorrect format    │",
    "│      ┆        ┆          ┆       ┆ */                        │",
    "╰──────┴────────┴──────────┴───────┴───────────────────────────╯",
    "╭──────┬────────┬──────────┬───────┬──────────────────────────╮",
    "│ Line ┆ Type   ┆ Due Date ┆ Owner ┆ python/test.py           │",
    "╞══════╪════════╪══════════╪═══════╪══════════════════════════╡",
    "│ 1    ┆ Format ┆          ┆       ┆ \"\"\"                      │",
    "│      ┆        ┆          ┆       ┆ TODO fix some issues     │",
    "│      ┆        ┆          ┆       ┆ \"\"\"                      │",
    "├╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┤",
    "│ 17   ┆ Format ┆          ┆       ┆ \"\"\"                      │",
    "│      ┆        ┆          ┆       ┆ TODO: incorrect format   │",
    "│      ┆        ┆          ┆       ┆ \"\"\"                      │",
    "├╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┤",
    "│ 27   ┆ Format ┆          ┆       ┆ # TODO: incorrect format │",
    "╰──────┴────────┴──────────┴───────┴──────────────────────────╯",
    "╭──────┬────────┬──────────┬───────┬──────────────────────────╮",
    "│ Line ┆ Type   ┆ Due Date ┆ Owner ┆ ruby/test.rb             │",
    "╞══════╪════════╪══════════╪═══════╪══════════════════════════╡",
    "│ 2    ┆ Format ┆          ┆       ┆ # TODO: incorrect format │",
    "├╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┤",
    "│ 8    ┆ Format ┆          ┆       ┆ =begin                   │",
    "│      ┆        ┆          ┆       ┆ TODO: incorrect format   │",
    "│      ┆        ┆          ┆       ┆ =end                     │",
    "╰──────┴────────┴──────────┴───────┴──────────────────────────╯",
    "╭──────┬────────┬──────────┬───────┬───────────────────────────╮",
    "│ Line ┆ Type   ┆ Due Date ┆ Owner ┆ rust/no_due.rs            │",
    "╞══════╪════════╪══════════╪═══════╪═══════════════════════════╡",
    "│ 2    ┆ Format ┆          ┆       ┆ // TODO: incorrect format │",
    "├╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┤",
    "│ 7    ┆ Format ┆          ┆       ┆ /*                        │",
    "│      ┆        ┆          ┆       ┆ TODO: incorrect format    │",
    "│      ┆        ┆          ┆       ┆ */                        │",
    "╰──────┴────────┴──────────┴───────┴───────────────────────────╯",
    "╭──────┬────────┬──────────┬───────┬───────────────────────────╮",
    "│ Line ┆ Type   ┆ Due Date ┆ Owner ┆ rust/test.rs              │",
    "╞══════╪════════╪══════════╪═══════╪═══════════════════════════╡",
    "│ 2    ┆ Format ┆          ┆       ┆ // TODO: incorrect format │",
    "├╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┤",
    "│ 8    ┆ Format ┆          ┆       ┆ /*                        │",
    "│      ┆        ┆          ┆       ┆ TODO: incorrect format    │",
    "│      ┆        ┆          ┆       ┆ */                        │",
    "╰──────┴────────┴──────────┴───────┴───────────────────────────╯",
    "╭──────┬────────┬──────────┬───────┬───────────────────────────╮",
    "│ Line ┆ Type   ┆ Due Date ┆ Owner ┆ typescript/test.ts        │",
    "╞══════╪════════╪══════════╪═══════╪═══════════════════════════╡",
    "│ 2    ┆ Format ┆          ┆       ┆ // TODO: incorrect format │",
    "├╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┤",
    "│ 8    ┆ Format ┆          ┆       ┆ /*                        │",
    "│      ┆        ┆          ┆       ┆ TODO: incorrect format    │",
    "│      ┆        ┆          ┆       ┆ */                        │",
    "╰──────┴────────┴──────────┴───────┴───────────────────────────╯",
]
